File: app/models/oqc.py
from sqlalchemy import Column, Integer, String, Text, TIMESTAMP, DECIMAL, ForeignKey
from sqlalchemy.orm import relationship
from app.database.base import Base

class OQC(Base):
    __tablename__ = "oqc"
    id = Column(Integer, primary_key=True, index=True)
    part_number = Column(String(100), ForeignKey("master_prod.part_number"))
    quantity_passed = Column(DECIMAL(10, 2), nullable=False)
    inspection_date = Column(TIMESTAMP)
    inspector_id = Column(Integer, ForeignKey("users.id"))
    notes = Column(Text)

    inspector = relationship("User")
    product = relationship("MasterProd")

File: node/models/oqc.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const OQC = sequelize.define('OQC', {
    id: { type: DataTypes.INTEGER, autoIncrement: true, primaryKey: true },
    part_number: { type: DataTypes.STRING(100), allowNull: false },
    quantity_passed: { type: DataTypes.DECIMAL(10, 2), allowNull: false },
    inspection_date: { type: DataTypes.DATE, defaultValue: DataTypes.NOW },
    inspector_id: { type: DataTypes.INTEGER },
    notes: { type: DataTypes.TEXT }
}, { tableName: 'oqc', timestamps: true });

module.exports = OQC;
